File: Programming Basics/06-Nested Loops - Exercise/sumPrimeNonPrime.js
function sumPrimeNonPrime(input) {

    let index = 0; 
    let theInput = input[index];
    index++
    let primeNumbers = 0;
    let notPrimeNumbers = 0;

    while(theInput !== "stop"){
        let numbers = Number(theInput);
        theInput = input[index];
        index++

        if(numbers < 0) {
            console.log("Number is negative.");
            continue;
        }

        if(numbers === 1) {
            notPrimeNumbers += numbers;
            continue;
        }

        let isItPossibleToBePrime = true;

        for(let i = 2; i < numbers; i++) {
            if (numbers % i === 0) {
                notPrimeNumbers += numbers;
                isItPossibleToBePrime = false;
                break;
            } 
        }

        if(isItPossibleToBePrime) {
            primeNumbers += numbers;
        }
    }

    console.log(`Sum of all prime numbers is: ${primeNumbers}`);
    console.log(`Sum of all non prime numbers is: ${notPrimeNumbers}`);
    
}

sumPrimeNonPrime(["0",
"-9",
"0",
"stop"])

File: Programming Basics Java/src/NestedLoopsExercise/SumPrimeNonPrime.java
package NestedLoopsExercise;

import java.util.Scanner;

public class SumPrimeNonPrime {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String input = scanner.nextLine();
        int sumOfPrimeNumbers = 0;
        int sumOfNonPrimeNumbers = 0;

        while (!input.equals("stop")) {
            int inputAsNumber = Integer.parseInt(input);

            if (inputAsNumber < 0) {
                System.out.println("Number is negative.");
                input = scanner.nextLine();
                continue;
            }

            boolean isPrime = true;

            for (int i = 2; i <= inputAsNumber / 2; i++) {
                if (inputAsNumber % i == 0) {
                    isPrime = false;
                    break;
                }
            }

            if (isPrime) {
                sumOfPrimeNumbers += inputAsNumber;
            } else {
                sumOfNonPrimeNumbers += inputAsNumber;
            }

            input = scanner.nextLine();
        }

        System.out.printf("Sum of all prime numbers is: %d\n", sumOfPrimeNumbers);
        System.out.printf("Sum of all non prime numbers is: %d", sumOfNonPrimeNumbers);
    }
}
